sdk/cosmos/azure-cosmos-spark_3-1_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala

```diff
@@ -117,19 +118,54 @@ private object CosmosAccountConfig {
   }
 }
 
-private case class CosmosReadConfig(forceEventualConsistency: Boolean)
+private case class CosmosReadConfig(forceEventualConsistency: Boolean,
+                                    jsonSchemaConversionMode: JsonSchemaConversionMode)
+
+private object JsonSchemaConversionModes extends Enumeration {
+  type JsonSchemaConversionMode = Value
+
+  val Relaxed: JsonSchemaConversionModes.Value = Value("Relaxed")
+  val Strict: JsonSchemaConversionModes.Value = Value("Strict")
+}
 
 private object CosmosReadConfig {
+  private val DefaultJsonSchemaConversionMode: JsonSchemaConversionMode = JsonSchemaConversionModes.Relaxed
+
   private val ForceEventualConsistency = CosmosConfigEntry[Boolean](key = "spark.cosmos.read.forceEventualConsistency",
     mandatory = false,
     defaultValue = Some(true),
     parseFromStringFunction = value => value.toBoolean,
     helpMessage = "Makes the client use Eventual consistency for read operations")
 
+  private val JsonSchemaConversion = CosmosConfigEntry[JsonSchemaConversionMode](
+    key = "spark.cosmos.read.jsonSchemaConversionMode",
+    mandatory = false,
+    defaultValue = Some(DefaultJsonSchemaConversionMode),
+    parseFromStringFunction = value => validateJsonSchemaConversion(value),
+    helpMessage = "Defines whether to throw on inconsistencies between schema definition and json attribute " +
+      "types (Strict) or to return null values (Relaxed).")
+
+  private def validateJsonSchemaConversion(mode: String): JsonSchemaConversionMode = {
+    Option(mode).fold(DefaultJsonSchemaConversionMode)(p => {
+      val modeName = p.trim
+
+      if (modeName.isEmpty) {
+        DefaultJsonSchemaConversionMode
```

Comment:
you are already setting the default above `defaultValue = Some(DefaultJsonSchemaConversionMode)`

can we drop the additional logic and rely on the `defaultValue` functionality of the CosmosConfigEntry?

Reply:
This handles the fact that the user is specifying the config but its an empty string

Reply:
then we should throw. that's invalid config